flow/layers/layer_tree.cc

```diff
@@ -52,9 +54,37 @@ bool LayerTree::Preroll(CompositorContext::ScopedFrame& frame,
       device_pixel_ratio_};
 
   root_layer_->Preroll(&context, frame.root_surface_transformation());
+
+  RasterCache(&context, frame.root_surface_transformation());
+
   return context.surface_needs_readback;
 }
 
+void LayerTree::RasterCache(PrerollContext* context, const SkMatrix& ctm) {
+  auto& need_raster_cache_layers = context->raster_cached_layers;
+  std::sort(
+      need_raster_cache_layers.begin(), need_raster_cache_layers.end(),
+      [](const Layer* a, const Layer* b) { return a->level() < b->level(); });
+
+  const int length = need_raster_cache_layers.size();
+  for (int i = 0; i < length; i++) {
+    auto* current_layer = need_raster_cache_layers[i];
+    // check the ancestor has raster cached
```

Comment:
RasterCache Layer will `painting` his children, may be can in this progress to flag the child has cached? https://github.com/flutter/engine/blob/main/flow/raster_cache.cc#L197